### PR TITLE
Enable cloning of `UnsupportedFeature` to allow use in image-tiff crate

### PR DESCRIPTION
**Pull request by AlanRace**
To solve https://github.com/image-rs/image-tiff/issues/168 I would like to clone the `UnsupportedFeature` error and add this to a `TiffError` to pass back to the caller.

### CODE REVIEW

**Review by fintelia — APPROVED** [review]

**Comment by micahsnyder** [comment]
It appears this is approved but hasn't merged. @HeroicKatora does this need your sign-off?

**Comment by HeroicKatora** [comment]
We usually wait for CI but it's not caused by this PR:

>       rust wasm file schema version: 0.2.82
>        this binary schema version: 0.2.80 (4caa98165)

**Pull request merged**
